server/controllers/socket.controller.ts

```diff
@@ -0,0 +1,197 @@
+import type { NextFunction, OpineRequest, OpineResponse } from "../deps.ts";
+import type { RoomSchema } from "../models/room.model.ts";
+import {
+	JoinEvent,
+	WebSocketEvent,
+	RoomUpdateEvent,
+	ConnectEvent,
+} from "../types/socket.ts";
+import connectToDb from "../utils/connectToDb.ts";
+import { lookupRoom } from "../utils/db.ts";
+
+const sockets = new Map<string, WebSocket>();
+export const { rooms, users } = await connectToDb();
+
+/**
+ * Adds a player to the room specified either as a moderator or voter.
+ * @param userId ID of the user joining
+ * @param data JoinEvent data
+ * @returns Whether the person who joined is the moderator of the room
+ */
+async function handleJoin(userId: string, data: JoinEvent): Promise<boolean> {
+	const roomData = await lookupRoom(data.roomCode);
+	if (!roomData) {
+		throw new Error(`No room with room code ${data.roomCode}.`);
+	}
+
+	const isModerator = !roomData.moderator;
```

Comment:
If I'm understanding, this basically means the first person to connect. This should work fine, as long as we destroy rooms after the last person leaves, so they are always created each time.
If we want to have rooms persist (so people can bookmark them, etc.), we will want to add more logic here.

Reply:
Yea that's a good point. People won't really be able to "bookmark" rooms with this strategy since we don't allow users to create rooms with their own room code. Definitely something to consider if we get a lot of feedback around this 🙂